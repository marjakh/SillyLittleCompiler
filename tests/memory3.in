let i = 0;
let j = 0;
let root_array = new Array(10);
let root_array_ix = 0;
while (i < 500) {
  let a = new Array(40);
  let k = 0;
  while (k < 40) {
    a[k] = i + k;
    k = k + 1;
  }
  i = i + 1;
  j = j + 1;
  if (j == 100) {
    root_array[root_array_ix] = a;
    root_array_ix = root_array_ix + 1;
    j = 0;
  }
}

root_array_ix = 0;
while (root_array_ix < 5) {
  let k = 0;
  while (k < 40) {
    write(root_array[root_array_ix][k]);
    k = k + 1;
  }
  root_array_ix = root_array_ix + 1;
}