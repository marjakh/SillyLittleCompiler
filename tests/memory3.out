99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
299
300
301
302
303
304
305
306
307
308
309
310
311
312
313
314
315
316
317
318
319
320
321
322
323
324
325
326
327
328
329
330
331
332
333
334
335
336
337
338
399
400
401
402
403
404
405
406
407
408
409
410
411
412
413
414
415
416
417
418
419
420
421
422
423
424
425
426
427
428
429
430
431
432
433
434
435
436
437
438
499
500
501
502
503
504
505
506
507
508
509
510
511
512
513
514
515
516
517
518
519
520
521
522
523
524
525
526
527
528
529
530
531
532
533
534
535
536
537
538
